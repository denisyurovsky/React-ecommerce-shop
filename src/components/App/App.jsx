import React from 'react';

import smile from '../../assets/images/smile.svg';

import styles from './App.module.scss';

export function App() {
  return (
    <div className={styles.container}>
      <div>
        <p data-testid="title" className={styles.hello}>
          Hi team&nbsp;
          <img src={smile} className={styles.smileIcon} alt="smile" />
        </p>
        <p>
          Today we are starting our course. This is a component-example
          describing a general structure of any component in our project. Please
          look at it, loot at the project structure and the structure of the
          component App, look at the file naming in JS and CSS code and try to
          follow these conventions. Also have a look at the App.test.js, it
          shows the general structure of a test file and basic testing
          principles.
        </p>
        <p>Good luck! :)</p>
        <a
          className="appLink"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </div>
    </div>
  );
}
