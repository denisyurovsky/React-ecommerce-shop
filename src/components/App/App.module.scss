.container {
  margin: 0 auto;
  width: 600px;
  text-align: center;
  font-size: 24px;
  display: flex;
  align-items: center;
}

.smileIcon {
  height: 32;
  width: 32px;
}

.hello {
  display: flex;
  justify-content: center;
}

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}
