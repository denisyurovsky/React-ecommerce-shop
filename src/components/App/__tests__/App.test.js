import { render, screen } from '@testing-library/react';
import React from 'react';

import { App } from '../App';

describe('App component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(<App />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('another group of tests', () => {
    beforeEach(() => {
      render(<App />);
    });

    it('renders title "Hi team"', () => {
      const title = screen.getByTestId('title');

      expect(title.textContent).toContain('Hi team');
    });

    it('renders "reactjs.org" as a link to the course', () => {
      const courseLink = screen.getByText(/learn react/i);

      expect(courseLink.getAttribute('href')).toBe('https://reactjs.org');
    });
  });
});
