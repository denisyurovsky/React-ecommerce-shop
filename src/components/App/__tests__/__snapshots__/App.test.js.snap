// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <p
        class="hello"
        data-testid="title"
      >
        Hi teamÂ 
        <img
          alt="smile"
          class="smileIcon"
          src="smile.svg"
        />
      </p>
      <p>
        Today we are starting our course. This is a component-example describing a general structure of any component in our project. Please look at it, loot at the project structure and the structure of the component App, look at the file naming in JS and CSS code and try to follow these conventions. Also have a look at the App.test.js, it shows the general structure of a test file and basic testing principles.
      </p>
      <p>
        Good luck! :)
      </p>
      <a
        class="appLink"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </div>
  </div>
</DocumentFragment>
`;
